llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -4177,7 +4177,9 @@ static bool isMaskOrZero(const Value *V, bool Not, const SimplifyQuery &Q,
 /// a check for a lossy truncation.
 /// Folds:
 ///   icmp SrcPred (x & Mask), x    to    icmp DstPred x, Mask
+///   icmp SrcPred (x & ~Mask), ~Mask    to    icmp DstPred x, ~Mask
 ///   icmp eq/ne (x & ~Mask), 0     to    icmp DstPred x, Mask
+///   icmp eq/ne (x | ~Mask), -1     to    icmp DstPred x, Mask
```

Comment:
Is this supposed to be?
```suggestion
///   icmp eq/ne (~x | Mask), -1     to    icmp DstPred x, Mask
```